File: ios/BackgroundSound.swift
import AVFoundation 

class BackgroundSound {
    private var player: AVPlayer?

    // Spela lokal fil från bundle
    func playBackgroundSound(filePath: String) {
        guard let url = URL(string: filePath) else {
            print("Invalid file URI: \(filePath)")
            return
        }

        // AVAudioSession för bakgrundsuppspelning
        do {
            try AVAudioSession.sharedInstance().setCategory(.playAndRecord, mode: .default, options: [.mixWithOthers, .defaultToSpeaker])
            try AVAudioSession.sharedInstance().setActive(true)
        } catch {
            print("Failed to set audio session category: \(error)")
        }

        // Skapa AVPlayer och spela
        let item = AVPlayerItem(url: url)
        player = AVPlayer(playerItem: item)
        player?.play()
        print("Playing background sound: \(filePath)")
    }

    func pauseBackgroundSound() {
        player?.pause()
        print("Paused background sound")
    }

    func resumeBackgroundSound() {
        player?.play()
        print("Resumed background sound")
    }

    func stopBackgroundSound() {
        player?.pause()
        player = nil
        print("Stopped background sound")
    }
}

File: android/src/main/java/com/margelo/nitro/decibel/BackgroundSound.kt
import android.content.Context
import android.net.Uri
import android.util.Log
import com.google.android.exoplayer2.ExoPlayer
import com.google.android.exoplayer2.MediaItem
import com.google.android.exoplayer2.Player

class BackgroundSound(private val context: Context) {

    private var player: ExoPlayer? = null

    /**
     * Spela ljudfil från URI (lokal fil eller nätverk)
     */
    fun playBackgroundSound(filePath: String) {
        stopBackgroundSound() // stoppa eventuell befintlig uppspelning

        try {
            val uri = Uri.parse(filePath)
            player = ExoPlayer.Builder(context).build().also { exo ->
                val mediaItem = MediaItem.fromUri(uri)
                exo.setMediaItem(mediaItem)
                exo.repeatMode = Player.REPEAT_MODE_OFF
                exo.playWhenReady = true
                exo.prepare()
            }

            Log.d("BackgroundSound", "Playing background sound: $filePath")
        } catch (e: Exception) {
            Log.e("BackgroundSound", "Failed to play background sound: $e")
        }
    }

    /**
     * Pausa uppspelningen
     */
    fun pauseBackgroundSound() {
        player?.pause()
        Log.d("BackgroundSound", "Paused background sound")
    }

    /**
     * Återuppta uppspelningen
     */
    fun resumeBackgroundSound() {
        player?.play()
        Log.d("BackgroundSound", "Resumed background sound")
    }

    /**
     * Stoppa och släpp resurser
     */
    fun stopBackgroundSound() {
        player?.release()
        player = null
        Log.d("BackgroundSound", "Stopped background sound")
    }
}
